specification/provisioningservices/resource-manager/Microsoft.Devices/2017-08-21-preview/iotdps.json

```diff
@@ -0,0 +1,1600 @@
+					"in": "query",
+					"required": false,
+					"type": "string"
+				}, {
+					"$ref": "#/parameters/api-version"
+				}],
+				"responses": {
+					"200": {
+						"description": "OK",
+						"schema": {
+							"$ref": "#/definitions/VerificationCodeResponse"
+						}
+					},
+					"default": {
+						"description": "DefaultErrorResponse",
+						"schema": {
+							"$ref": "#/definitions/ErrorDetails"
+						}
+					}
+				},
+				"deprecated": false
+			}
+		},
+		"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Devices/provisioningServices/{provisioningServiceName}/certificates/{certificateName}/verify": {
+			"post": {
+				"tags": ["POST"],
+				"operationId": "DpsCertificate_VerifyCertificate",
+				"consumes": ["application/json"],
+				"produces": ["application/json"],
+				"parameters": [{
+					"name": "certificateName",
+					"in": "path",
+					"required": true,
+					"type": "string"
+				}, {
+					"name": "request",
+					"in": "body",
+					"required": true,
+					"schema": {
+						"$ref": "#/definitions/VerificationCodeRequest"
+					}
+				}, {
+					"$ref": "#/parameters/subscriptionId"
+				}, {
+					"name": "resourceGroupName",
+					"in": "path",
+					"required": true,
+					"type": "string"
+				}, {
+					"name": "provisioningServiceName",
+					"in": "path",
+					"required": true,
+					"type": "string"
+				}, {
+					"name": "certificate.name",
+					"in": "query",
+					"required": false,
+					"type": "string"
+				}, {
+					"name": "certificate.rawBytes",
+					"in": "query",
+					"required": false,
+					"type": "string",
+					"format": "byte"
+				}, {
+					"name": "certificate.isVerified",
+					"in": "query",
+					"required": false,
+					"type": "boolean"
+				}, {
+					"name": "certificate.purpose",
+					"in": "query",
+					"required": false,
+					"type": "string",
+					"enum": ["clientAuthentication", "serverAuthentication"]
+				}, {
+					"name": "certificate.created",
+					"in": "query",
+					"required": false,
+					"type": "string",
+					"format": "date-time"
+				}, {
+					"name": "certificate.lastUpdated",
+					"in": "query",
+					"required": false,
+					"type": "string",
+					"format": "date-time"
+				}, {
+					"name": "certificate.hasPrivateKey",
+					"in": "query",
+					"required": false,
+					"type": "boolean"
+				}, {
+					"name": "certificate.nonce",
+					"in": "query",
+					"required": false,
+					"type": "string"
+				}, {
+					"$ref": "#/parameters/api-version"
+				}],
+				"responses": {
+					"200": {
+						"description": "OK",
+						"schema": {
+							"$ref": "#/definitions/CertificateResponse"
+						}
+					},
+					"default": {
+						"description": "DefaultErrorResponse",
+						"schema": {
+							"$ref": "#/definitions/ErrorDetails"
+						}
+					}
+				},
+				"deprecated": false
+			}
+		},
+		"/subscriptions/{subscriptionId}/providers/Microsoft.Devices/checkProvisioningServiceNameAvailability": {
```

Comment:
why not just call it checkNameAvailability as described in the RPC. the resource type comes in the request payload.

Reply:
We have different RPs for reach resource type but ARM manifest doesn't have way to use same API for different resource types going to different RP endpoint. I spoke to Gaurav from ARM team, he is going think about it. Until then he agreed to use a different API

Reply:
@shankarsama - yes, you spoke to me. And I didnt realize its the same RP we had discussed :).